src/ansible_navigator/configuration_subsystem/navigator_post_processor.py

```diff
@@ -429,6 +429,37 @@ def exec_shell(
         """
         return self._true_or_false(entry, config)
 
+    @_post_processor
+    def help_builder(
+        self,
+        entry: SettingsEntry,
+        config: ApplicationConfiguration,
+    ) -> PostProcessorReturn:
+        # pylint: disable=unused-argument
+        """Post process help_builder"""
+        messages, exit_messages = self._true_or_false(entry, config)
+        if all(
+            (
+                entry.value.current is True,
+                config.app == "builder",
+                config.mode == "interactive",
+            )
+        ):
```

Comment:
(important)
1) This is not a good use for `all()`. There is no need to create an iterable from already pre-calculated values only for "all" to iterate over them. It's actually less performant and consumes more memory than an alternative with an early exist which is to use condition chaining with an `and`.
2) This is so big and nested that it's hard to digest when embedded in a huge if-clause. It's better to assign a meaningful semantically significant name to the whole complex condition and maybe to its sub-conditions too. I tried to do some guessing what the meaning is below (it's probably wrong but it's the best one can do when reading this patch within the provided context):
    ```suggestion
            params_need_checking = (
                    entry.value.current is True and
                    config.app == "builder" and
                    config.mode == "interactive"
            )
            if params_need_checking:
    ```
    ```suggestion
            entry_value_is_selected = entry.value.current is True
            requested_app_is_builder = config.app == "builder"
            runtime_is_interactive = config.mode == "interactive"
            an_interactive_builder_cmd_is_selected = (
                entry_value_is_selected and
                requested_app_is_builder and
                runtime_is_interactive
            )
            if an_interactive_builder_cmd_is_selected:
    ```

Tell me if I guessed any of these correctly :)

Reply:
Yes you guessed it correctly :-). 

Nice suggestion. Have fixed it now.